var/spack/repos/builtin/packages/truchas/package.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Truchas(CMakePackage):
+    """Physics-based modeling and simulation of manufacturing processes.
+
+    Truchas includes coupled physics models for incompressible multi-material
+    flow with interface tracking, heat transfer, phase change, view factor
+    thermal radiation, species advection-diffusion, elastic/plastic mechanics
+    with contact, and electromagnetics. It employs finite volume, finite element,
+    and mimetic finite difference discretizations on 3-D unstructured meshes
+    composed of mixed cell types.
+    """
+
+    homepage = "https://www.truchas.org"
+    url      = "https://gitlab.com/truchas/truchas/-/archive/22.04/truchas-22.04.tar.bz2"
+    git      = "https://gitlab.com/truchas/truchas.git"
+
+    maintainers = ['pbrady']
+
+    version('develop', branch="master")
+    version('22.04', sha256="a2bc074d724b8f8752b44633b2f0e3ed0a840e01fc940510200641522040e6e3")
+
+    # ------------------------------------------------------------ #
+    # Variants
+    # ------------------------------------------------------------ #
+
+    variant("portage", default=False, description="use the portage data mapping tool")
+    variant("metis", default=True, description="use metis for grid partitioning")
+    variant("std_name", default=False, description="enable std_mod_proc_name with intel")
+    variant("config", default=True, description="use proved truchas config files for cmake")
+
+    # ------------------------------------------------------------ #
+    # Build dependencies
+    # ------------------------------------------------------------ #
+    depends_on('cmake@3.16:', type='build')
+
+    # ------------------------------------------------------------ #
+    # Test suite and restart utils
+    # ------------------------------------------------------------ #
+    depends_on('python@3.5:')
+    depends_on('py-numpy@1.12:')
+    depends_on('py-h5py@2.6.0')
+
+    # ------------------------------------------------------------ #
+    # IO dependencies
+    # ------------------------------------------------------------ #
+    depends_on('exodusii@2021-04-05+mpi')
+    depends_on('scorpio ^hdf5@1.10.6')
+    depends_on('petaca@22.03: +shared')
+    depends_on('petaca@22.03: +shared +std_name', when='+std_name')
+
+    # ------------------------------------------------------------ #
+    # Partitioning
+    # ------------------------------------------------------------ #
+    depends_on('chaco')
+    depends_on('metis@5.1.0', when='+metis')
```

Comment:
We usually don't pin exact versions, since they might cause conflicts in environments. Is it really necessary to use only `v5.1.0`?

Reply:
Thanks for the review @alalazo.  I've relaxed the version requirements so we don't pin exact versions anymore.